### PR TITLE
Remove support for Swift 5.1

### PR DESCRIPTION
**Pull request by Davidde94**
Changes made:

Removed CI config for 5.1
Changed 5.2 to be based on 16.04 so we have some coverage there
Converted Package.swift to the latest syntax
Added a description to supported Swift versions in the README

### CODE REVIEW

**Comment by Davidde94** [comment]
Hey @yim-lee some more CI playing if you don't mind! Could you please:

- [ ]   Remove the 5.1 jobs
- [ ]   Update the 5.2 job to use 16.04
- [ ]   Update the API breakage to be based on the Swift 5.4 Ubuntu 20.04 image

**Review by Lukasa — APPROVED** [review]

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@Davidde94 CI changes are done.

- This project doesn't have api-breakage job :)
- Added 5.4 job to required list
- Fixed 5.5 job
- 5.2 and soundness failures look legit

**Comment by Davidde94** [comment]
@yim-lee it looks like this is a failure in TSan itself. Could we try swapping the 5.3 job to 16.04, and the 5.2 job to 18.04. I've pushed up the necessary files.

File: docker/docker-compose.1604.53.yaml [hunk 1]
```diff
@@ -3,17 +3,17 @@ version: "3"
...
     build:
       args:
-        ubuntu_version: "bionic"
+        ubuntu_version: "xenial"
         swift_version: "5.3"
 
   test:
-    image: swift-nio-ssh:18.04-5.3
+    image: swift-nio-ssh:16.04-5.3
     environment: []
```
Comment (yim-lee) [inline docker/docker-compose.1604.53.yaml:14]: Need to remove `[]`

Comment (Davidde94) [reply docker/docker-compose.1604.53.yaml]: Thanks! I've fixed it

**Comment by Davidde94** [comment]
@yim-lee could you take another look when you get the chance please?

**Comment by yim-lee** [comment]
> it looks like this is a failure in TSan itself. Could we try swapping the 5.3 job to 16.04, and the 5.2 job to 18.04. I've pushed up the necessary files.

@Davidde94 done.

@swift-server-bot test this please

**Comment by Davidde94** [comment]
Same TSan failure. For now I'll disable the job, we know that it's unreliable.

**Comment by yim-lee** [comment]
> Same TSan failure. For now I'll disable the job, we know that it's unreliable.

@Davidde94 Are there CI changes you want me to make?

**Comment by Davidde94** [comment]
@yim-lee I think this is something I can do myself, I'll push up now

**Pull request merged**